src/lib/pmd/PmdWrapper.ts

```diff
@@ -68,7 +72,9 @@ export default class PmdWrapper extends PmdSupport {
 		// NOTE: If we were going to run this command from the CLI directly, then we'd wrap the classpath in quotes, but this
 		// is intended for child_process.spawn(), which freaks out if you do that.
 		const classpath = await super.buildClasspath();
-		const args = ['-cp', classpath.join(path.delimiter), HEAP_SIZE, MAIN_CLASS, '-dir', this.path,
+		this.tempFile = tmp.fileSync();
+		fs.writeFileSync(this.tempFile.name, this.path);
```

Comment:
We've been following SFDX team's best practice to prefer async IO operations over sync so that Node's resources are used to the fullest. Could you convert these to async as well?